packages/canary-features/addon/index.js

```diff
@@ -1 +1,32 @@
-// export const sample_feature_flag = true;
+import { assign } from '@ember/polyfills';
+import global from '@ember-data/canary-features/environment';
+// import { ENV } from '@ember-data/canary-features/
+export const ENV = {
+  FEATURES: {},
+};
+console.log(global.EMBER_ENV, global.ENV, global.FEATURES);
+(EmberENV => {
+  if (typeof EmberENV !== 'object' || EmberENV === null) return;
+
+  for (let flag in EmberENV) {
+    if (
+      !EmberENV.hasOwnProperty(flag) ||
+      flag === 'EXTEND_PROTOTYPES' ||
+      flag === 'EMBER_LOAD_HOOKS'
+    )
+      continue;
+    let defaultValue = ENV[flag];
+    if (defaultValue === true) {
+      ENV[flag] = EmberENV[flag] !== false;
+    } else if (defaultValue === false) {
+      ENV[flag] = EmberENV[flag] === true;
+    }
+  }
+})(global.EmberENV || global.ENV);
+
+export const DEFAULT_FEATURES = {
+  sample_feature_flag: null,
+};
+
+export const FEATURES = assign(DEFAULT_FEATURES, ENV.FEATURES);
```

Comment:
You don’t want to mutate DEFAULT_FEATURES here, probably want `assign({}, DEFAULT_FEATURES, ENV.FEATURES)`